Clarify how to pin the Sphinx version (#8687)
readthedocs/readthedocs.org: docs/config-file/v2.rst

     configuration: conf.py
     fail_on_warning: true

.. note::

>    If you want to pin Sphinx to a specific version,
>    use a ``requirements.txt`` or ``environment.yml`` file

Reviewer: ```suggestion

   It is recommended to pin Sphinx to a specific version using a ``requirements.txt`` or ``environment.yml`` file
```

Reviewer: We should also mention this for MkDocs

Author: Good catch! Adding

   (see :ref:`config-file/v2:requirements file` and  :ref:`config-file/v2:conda.environment`).
   This also allows you to override :ref:`the default pinning done by Read the Docs

>    if your project was created before October 2020 <builds:external dependencies>`.

Reviewer 2: I'd really like to have a doc dedicated to pinning dependencies that covers all of the nice patterns we support. This would be a good doc to go over nice packaging patterns like extra requires and using poetry.

Our config docs are the best we have right now though, but it is a bit obtuse from a user perspective.

Also most users just listen to our advice, so telling users to "use a requirements.txt file" is a bit hostile to package maintainers, which don't use this concept in packaging. We've had this push back from users before, where package maintainers want to use concepts they are already using in packaging, like extra requires.

Not something that needs to hold up this PR, but it would be nice to return to this later, change the language to be a bit more general, and point to a feature page dedicated to pinning dependencies. Pinning dependencies is a common enough part of our configuration, and users have enough problems understanding it, that we really should have top level docs on this.

Author: I see your point about requirements files. As a library maintainer, I don't want to duplicate dependencies in a metadata file with abstract dependencies plus a list of concrete dependencies. (On abstract vs concrete: Donald Stufft explained everything here https://caremad.io/posts/2013/07/setup-vs-requirement/)

I will tweak the language to make it more amenable for other users, while making it clear that concrete dependencies are better suited for folks publishing docs that are not read from a reusable Python library.
